cola/widgets/startup.py

```diff
@@ -68,46 +76,52 @@ def __init__(self, context, parent=None):
 
         # Bookmarks/"Favorites" and Recent are lists of {name,path: str}
         settings = context.settings
-        bookmarks = settings.bookmarks
-        recent = settings.recent
-        all_repos = bookmarks + recent
+        self._setup_model(bookmarks_model, bookmark_items, settings.bookmarks)
+        self._setup_model(recent_model, recent_items, settings.recent)
+
+        self.bookmarks = bookmarks = BookmarksListView(
+            LIST_VIEW_BOOKMARK,
+            context,
+            bookmarks_model,
+            self.open_selected_bookmark,
+            self.set_model,
+        )
+        self.recent = recent = BookmarksListView(
+            LIST_VIEW_RECENT,
+            context,
+            recent_model,
+            self.open_selected_recent,
+            self.set_model,
+        )
 
-        directory_icon = icons.directory()
-        user_role = Qt.UserRole
-        normalize = display.normalize_path
-        paths = {normalize(repo['path']) for repo in all_repos}
-        short_paths = display.shorten_paths(paths)
-        self.short_paths = short_paths
+        self.recent_title_label = qtutils.label(
+            text="Recent", align=Qt.AlignHCenter | Qt.AlignVCenter, selectable=False
+        )
+        title_font = self.recent_title_label.font()
+        title_font.setBold(True)
+        title_font.setPixelSize(LABEL_TITLE_FONT_SIZE)
```

Comment:
Instead of using a hard-coded font size, lookup the current font size and then scale by some factor (eg. `0.8`). Is the default size too small or too large?

Reply:
That is the coolest way and I've never come up with it.
Thank you! I'll improve it.